Source/WebInspectorUI/UserInterface/Views/Popover.js

```diff
@@ -49,6 +49,14 @@ WI.Popover = class Popover extends WI.Object
         this._container.className = "container";
 
         this._drawBackgroundAnimationIdentifier = undefined;
+
+        // Background canvas
+        this._canvas = document.createElement("canvas");
```

Comment:
this will cause us to create a `<canvas>` each time we create a `WI.Popover`, whereas with `-webkit-canvas` there was only ever one.  in order to preserve the same behavior we need to make this shared among all `WI.Popover` (e.g. make some `static _getCanvas(width, height)` or something, maybe even using a `WeakRef` so that it's not kept alive longer than it needs to be if the `<canvas>` is removed from the DOM and all `WI.Popover` are destroyed)

Reply:
> this will cause us to create a <canvas> each time we create a WI.Popover

@dcrousso Isn't that expected if multiple popovers are shown at the same time (if that's possible)? Otherwise, only one of them will have a visible background

Reply:
> > this will cause us to create a  each time we create a WI.Popover
> 
> @dcrousso Isn't that expected if multiple popovers are shown at the same time (if that's possible)? Otherwise, only one of them will have a visible background

We should only ever show a single popover, matching the Cocoa platform paradigm for popovers. Anything that causes multiple popovers to appear simultaneously would be a bug, and also wouldn't work today with the shared canvas via the webkit-prefixed CSS property.

Reply:
yeah, it's not so much the concern about there being more than one `WI.Popover` as creating a _new_ `<canvas>` for _each_ `WI.Popover` (whereas with `-webkit-canvas` there was only every a _single_ `<canvas>` that was shared by _all_ `WI.Popover`)

even if every `WI.Popover` is super tiny (e.g. showing the value of a CSS variable), that's still potentially a _lot_ of `<canvas>` we'd be creating and destroying, which is probably not super great churn

Reply:
Regarding to the problem "replace getCSSCanvasContext with canvas": my proposal is to create a canvas element (and add it to body element) when the first "popover" is created. When the popover is displayed "this canvas" element can be moved to popover element, and when popover is hidden then move back to body element(to have it all the time and not create each time). What do you think? I am preparing a change with this solution.

Reply:
oof that's no bueno.  we probably shouldn't allow for that 😅  @patrickangle can you file a bug to fix that?  my guess is that it's cause we only listen for `"mousedown"` to dismiss the `WI.Popover`, whereas the `WI.CodeMirrorTokenTrackingController` looks at `"mousemove"` (or something like that)

@pgorszkowski-igalia i would prefer that we don't keep the `<canvas>` alive by adding it to the `<body>` when there are no `WI.Popover` that actively need it.  how about something like this instead?
```
static _getCanvasContext(width, height)
{
    let context = WI.Popover._canvasContext?.deref();
    if (!context) {
        context = document.createElement("canvas").getContext("2d");
        WI.Popover._canvasContext = new WeakRef(context);
    }

    context.canvas.width = width;
    context.canvas.height = height;
    return context;
}   
```
which could then be used like so
```
let context = WI.Popover._getCanvasContext(scaledWidth, scaledHeight);
this._element.appendChild(context.canvas);
```
tho we may wanna do this with different timing so that we're not calling `WI.Popover._getCanvasContext` or appending the `<canvas>` multiple times for the same `WI.Popover` 🤔

Reply:
I filed a bug about multiple popovers: https://bugs.webkit.org/show_bug.cgi?id=253557